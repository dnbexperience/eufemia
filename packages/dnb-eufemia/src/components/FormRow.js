/**
 * ATTENTION: This file is auto generated by using "prepareTemplates".
 * Do not change the content!
 *
 */

/**
 * Library Index form-row to autogenerate all the components and patterns
 * Used by "prepareFormRows"
 */

import FormRow from './form-row/FormRow'
export * from './form-row/FormRow'
export default FormRow
