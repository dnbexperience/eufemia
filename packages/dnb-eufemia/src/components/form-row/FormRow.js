/**
 * Web FormRow Component
 *
 * This is a legacy component.
 * For referencing while developing new features, please use a Functional component.
 */

import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import {
  extendGracefully,
  extendPropsWithContextInClassComponent,
  isTrue,
  makeUniqueId,
  validateDOMAttributes,
} from '../../shared/component-helper'
import AlignmentHelper from '../../shared/AlignmentHelper'

import Context from '../../shared/Context'
import FormLabel from '../form-label/FormLabel'
import Section from '../section/Section'
import {
  spacingPropTypes,
  createSpacingClasses,
} from '../space/SpacingHelper'

export const formRowPropTypes = {
  id: PropTypes.string,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.node,
  ]),
  label_direction: PropTypes.oneOf(['vertical', 'horizontal']),
  label_sr_only: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  label_id: PropTypes.string,
  label_class: PropTypes.string,
  no_label: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  no_fieldset: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  locale: PropTypes.string,
  wrap: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  direction: PropTypes.oneOf(['vertical', 'horizontal']),
  vertical: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  centered: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  section_style: PropTypes.string,
  section_spacing: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  globalStatus: PropTypes.shape({
    id: PropTypes.string,
    message: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  }),
  responsive: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  disabled: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  skeleton: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),

  skipContentWrapperIfNested: PropTypes.bool,

  ...spacingPropTypes,

  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.node,
  ]),
}

export const formRowDefaultProps = {
  id: null,
  label: null,
  label_direction: null,
  label_sr_only: null,
  label_id: null,
  label_class: null,
  no_label: false,
  no_fieldset: null,
  locale: null,
  wrap: null,
  direction: null,
  vertical: null,
  centered: null,
  section_style: null,
  section_spacing: null,
  globalStatus: null,
  responsive: null,
  disabled: null,
  skeleton: null,

  skipContentWrapperIfNested: false,
  className: null,
  children: null,
}

export default class FormRow extends React.PureComponent {
  static contextType = Context

  static propTypes = {
    ...formRowPropTypes,
  }

  static defaultProps = {
    ...formRowDefaultProps,
  }

  static getContent(props) {
    let label = null
    let children =
      typeof props.children === 'function'
        ? props.children(props)
        : props.children

    if (Array.isArray(props.children)) {
      children = children.reduce((pV, cV) => {
        if (cV && cV.type && cV.type.name === 'FormLabel') {
          label = cV.props.children
        } else {
          pV.push(cV)
        }
        return pV
      }, [])
    }

    return { label, children }
  }

  constructor(props) {
    super(props)
    this._id = props.id || makeUniqueId() // cause we need an id anyway

    // We may consider to use this later to check if we are inside FormSet
    // this.isInsideFormSet =
    //   context.FormRow && context.FormRow.isInsideFormSet
  }

  render() {
    // use only the props from context, who are available here anyway
    const props = extendPropsWithContextInClassComponent(
      this.props,
      FormRow.defaultProps,
      this.context?.FormRow, // nested FormRow
      this.context?.formElement // nested formElement
    )

    const {
      label_direction,
      label_sr_only,
      label_id,
      label_class,
      no_fieldset,
      no_label,
      locale,
      direction,
      vertical,
      centered,
      section_style,
      section_spacing,
      globalStatus,
      responsive,
      disabled,
      skeleton,
      wrap,
      className,
      skipContentWrapperIfNested,

      id: _id, // eslint-disable-line
      label: _label, // eslint-disable-line

      ...attributes
    } = props

    let { label } = props

    const isNested =
      this.context?.FormRow?.itsMeAgain ||
      this.context?.formElement?.itsMeAgain

    // in case we have a label already, we split this out and use this one instead
    const { label: nestedLabel, children } = FormRow.getContent(this.props)
    if (!label && nestedLabel) {
      label = nestedLabel
    }
    const hasLabel =
      (typeof label === 'string' && label.length > 0) || label
        ? true
        : false

    const id = this._id
    const params = {
      className: classnames(
        'dnb-form-row',
        (isTrue(vertical) || direction) &&
          `dnb-form-row--${isTrue(vertical) ? 'vertical' : direction}`,
        (isTrue(vertical) || label_direction) &&
          `dnb-form-row--${
            isTrue(vertical) ? 'vertical' : label_direction
          }-label`, // <-- has label
        centered && 'dnb-form-row--centered',
        isNested && 'dnb-form-row--nested',
        createSpacingClasses(props),
        className
      ),
      ...attributes,
    }

    // also used for code markup simulation
    validateDOMAttributes(this.props, params)

    const formElement = {
      useId: () => {
        if (this.isIsUsed) {
          // make a new ID, as we used one
          return makeUniqueId() // cause we need an id anyway
        }
        this.isIsUsed = true
        return id
      },
      itsMeAgain: true,
      hasLabel,
      globalStatus,
      direction,
      vertical,
      label_direction: isTrue(vertical) ? 'vertical' : label_direction,
      responsive,
      disabled,
      skeleton,
    }

    const providerContext = extendGracefully(this.context, {
      locale: locale ? locale : this.context.locale,
      formElement,
      FormRow: formElement,
    })

    const useFieldset = !isTrue(no_fieldset) && hasLabel

    return (
      <Context.Provider value={providerContext}>
        <Fieldset useFieldset={useFieldset}>
          <Section
            style_type={section_style || 'transparent'}
            spacing={section_spacing}
            element="div"
            {...params}
          >
            <AlignmentHelper />

            {label && (
              <FormLabel
                className={classnames('dnb-form-row__label', label_class)}
                id={label_id ? label_id : id + '-label'}
                for_id={useFieldset ? null : id} // we don't use for_id, because we don't have a single element to target to
                text={label}
                element={useFieldset ? 'legend' : 'label'}
                label_direction={label_direction}
                sr_only={label_sr_only}
                disabled={disabled}
                skeleton={skeleton}
              />
            )}

            {isTrue(no_label) && (
              <span
                className="dnb-form-label dnb-form-row__label-dummy"
                aria-hidden
              />
            )}

            {isNested && skipContentWrapperIfNested ? (
              children
            ) : (
              <div
                className={classnames(
                  'dnb-form-row__content',
                  isTrue(wrap) && 'dnb-form-row__content--wrap',
                  label && !isTrue(vertical) && direction !== 'vertical',
                  responsive && 'dnb-responsive-component'
                )}
              >
                {children}
              </div>
            )}
          </Section>
        </Fieldset>
      </Context.Provider>
    )
  }
}

const Fieldset = ({
  useFieldset,
  children,
  className = null,
  ...props
}) => {
  if (useFieldset) {
    return (
      <fieldset
        className={classnames('dnb-form-row__fieldset', className)}
        {...props}
      >
        {children}
      </fieldset>
    )
  }
  return (
    <div
      className={classnames('dnb-form-row__fieldset', className)}
      {...props}
    >
      {children}
    </div>
  )
}

// docs (or use ptd): https://github.com/facebook/prop-types#usage
Fieldset.propTypes = {
  children: PropTypes.node,
  useFieldset: PropTypes.bool,
  className: PropTypes.string,
}
Fieldset.defaultProps = {
  children: null,
  useFieldset: false,
  className: null,
}

FormRow._supportsSpacingProps = true
