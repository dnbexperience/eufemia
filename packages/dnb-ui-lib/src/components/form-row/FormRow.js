/**
 * Web FormRow Component
 *
 */

import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import AlignmentHelper from '../../shared/AlignmentHelper'
import {
  extend,
  extendPropsWithContext,
  isTrue,
  makeUniqueId,
  registerElement,
  validateDOMAttributes
  // processChildren
} from '../../shared/component-helper'
import Context from '../../shared/Context'
import hashSum from '../../shared/libs/HashSum'
import FormLabel from '../form-label/FormLabel'
import { createSpacingClasses } from '../space/SpacingHelper'

const renderProps = {
  render_content: null
}

export const propTypes = {
  id: PropTypes.string,
  label: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.node
  ]),
  label_direction: PropTypes.oneOf(['vertical', 'horizontal']),
  label_sr_only: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  label_id: PropTypes.string,
  label_class: PropTypes.string,
  no_label: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  no_fieldset: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  indent: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  wrap: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  direction: PropTypes.oneOf(['vertical', 'horizontal']),
  vertical: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  indent_offset: PropTypes.string,
  section_style: PropTypes.string,
  section_spacing: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  global_status_id: PropTypes.string,
  disabled: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  class: PropTypes.string,

  /** React props */
  skipContentWrapperIfNested: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.func,
    PropTypes.node
  ]),

  // Web Component props
  custom_element: PropTypes.object,
  custom_method: PropTypes.func,
  render_content: PropTypes.func
}

const defaultProps = {
  id: null,
  label: null,
  label_direction: null,
  label_sr_only: null,
  label_id: null,
  label_class: null,
  no_label: false,
  no_fieldset: null,
  indent: null,
  wrap: null,
  direction: null,
  vertical: null,
  indent_offset: null,
  section_style: null,
  section_spacing: null,
  global_status_id: null,
  disabled: null,
  class: null,

  /** React props */
  skipContentWrapperIfNested: false,
  className: null,
  children: null,

  // Web Component props
  custom_element: null,
  custom_method: null,
  ...renderProps
}

export default class FormRow extends PureComponent {
  static tagName = 'dnb-form-row'
  static propTypes = propTypes
  static defaultProps = defaultProps
  static contextType = Context

  static enableWebComponent() {
    registerElement(FormRow.tagName, FormRow, defaultProps)
  }

  static getContent(props) {
    if (typeof props.render_content === 'function')
      props.render_content(props)

    let label = null
    let children =
      typeof props.children === 'function'
        ? props.children(props)
        : props.children

    if (Array.isArray(props.children)) {
      children = children.reduce((pV, cV) => {
        if (cV.type && cV.type.name === 'FormLabel') {
          label = cV.props.children
        } else {
          pV.push(cV)
        }
        return pV
      }, [])
    }

    return { label, children }
  }

  constructor(props) {
    super(props)
    this._id = props.id || makeUniqueId() // cause we need an id anyway

    // Not used yet
    // this.isInsideFormSet =
    //   context.formRow && context.formRow.isInsideFormSet
  }

  render() {
    // use only the props from context, who are available here anyway
    const props = extendPropsWithContext(
      this.props,
      defaultProps,
      this.context.formRow
    )

    let {
      label,
      label_direction,
      label_sr_only,
      label_id,
      label_class,
      no_fieldset,
      no_label,
      indent,
      direction,
      vertical,
      indent_offset,
      section_style,
      section_spacing,
      global_status_id,
      disabled,
      wrap,
      id: _id, // eslint-disable-line
      className,
      class: _className,
      skipContentWrapperIfNested,

      ...attributes
    } = props

    const isNested =
      this.context.formRow && this.context.formRow.itsMeAgain

    // in case we have a label already, we split this out and use this one instead
    const { label: nestedLabel, children } = FormRow.getContent(this.props)
    if (!label && nestedLabel) {
      label = nestedLabel
    }
    const hasLabel =
      (typeof label === 'string' && label.length > 0) || label
        ? true
        : false

    const id = this._id
    const params = {
      className: classnames(
        'dnb-form-row',
        (isTrue(vertical) || direction) &&
          `dnb-form-row--${isTrue(vertical) ? 'vertical' : direction}`,
        (isTrue(vertical) || label_direction) &&
          `dnb-form-row--${
            isTrue(vertical) ? 'vertical' : label_direction
          }-label`, // <-- has label
        indent &&
          !(
            isNested &&
            this.context.formRow.hasLabel &&
            this.context.formRow.indent
          ) &&
          `dnb-form-row__indent--${isTrue(indent) ? 'default' : indent}`,
        isNested && 'dnb-form-row--nested',
        section_style ? `dnb-section dnb-section--${section_style}` : null,
        section_spacing
          ? `dnb-section--spacing-${
              isTrue(section_spacing) ? 'default' : section_spacing
            }`
          : null,
        createSpacingClasses(props),
        className,
        _className
      ),
      ...attributes
    }

    // also used for code markup simulation
    validateDOMAttributes(this.props, params)

    // check if context has changed, if yes, then update the cache
    if (hashSum(this._cachedContext) !== hashSum(this.context)) {
      this._cachedContext = this.context

      const formRow = {
        useId: () => {
          if (this.isIsUsed) {
            // make a new ID, as we used one
            return makeUniqueId() // cause we need an id anyway
          }
          this.isIsUsed = true
          return id
        },
        itsMeAgain: true,
        hasLabel,
        indent,
        global_status_id,
        direction,
        vertical,
        label_direction: isTrue(vertical) ? 'vertical' : label_direction,
        disabled
      }
      this._contextWeUse = extend(this.context, {
        formRow
      })
    }

    const useFieldset = !isTrue(no_fieldset) && hasLabel

    return (
      <Context.Provider value={this._contextWeUse}>
        <Fieldset useFieldset={useFieldset}>
          <div {...params}>
            <AlignmentHelper className="dnb-form-row__helper" />

            {label && (
              <FormLabel
                className={classnames('dnb-form-row__label', label_class)}
                id={label_id ? label_id : id + '-label'}
                for_id={useFieldset ? null : id} // we don't use for_id, because we don't have a single element to target to
                text={label}
                element={useFieldset ? 'legend' : 'label'}
                direction={label_direction}
                sr_only={label_sr_only}
                disabled={isTrue(disabled)}
              />
            )}

            {isTrue(no_label) && (
              <span
                className="dnb-form-label dnb-form-row__label-dummy"
                aria-hidden
              />
            )}

            {isNested && skipContentWrapperIfNested ? (
              children
            ) : (
              <div
                className={classnames(
                  'dnb-form-row__content',
                  isTrue(wrap) && 'dnb-form-row__content--wrap',
                  label &&
                    !isTrue(vertical) &&
                    direction !== 'vertical' &&
                    `dnb-form-row__content--${indent_offset || 'default'}`
                )}
                ref={this._contentRef}
              >
                {children}
              </div>
            )}
          </div>
        </Fieldset>
      </Context.Provider>
    )
  }
}

const Fieldset = ({ useFieldset, className, children, ...props }) => {
  if (useFieldset) {
    return (
      <fieldset
        className={classnames('dnb-form-row__fieldset', className)}
        {...props}
      >
        {children}
      </fieldset>
    )
  }
  return (
    <div
      className={classnames('dnb-form-row__fieldset', className)}
      {...props}
    >
      {children}
    </div>
  )
}

// docs (or use ptd): https://github.com/facebook/prop-types#usage
Fieldset.propTypes = {
  children: PropTypes.node.isRequired,
  useFieldset: PropTypes.bool,
  className: PropTypes.string
}
Fieldset.defaultProps = {
  useFieldset: false,
  className: null
}

export const prepareFormRowContext = props => {
  if (typeof props.label_direction === 'undefined') {
    props.label_direction = isTrue(props.vertical)
      ? 'vertical'
      : props.label_direction
  }
  return props
}
