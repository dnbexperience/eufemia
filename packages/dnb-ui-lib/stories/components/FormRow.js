/**
 * dnb-ui-lib Component Story
 *
 */

import React /* , { useState, useEffect } */ from 'react'
import { Wrapper, Box } from '../helpers'
import styled from '@emotion/styled'

import { H2, H3 } from '../../src/elements'
import AllComponents from '../../src/components/form-row/AllComponents'

// import Provider from '../../src/shared/Provider'
import {
  Checkbox,
  Radio,
  Input,
  FormLabel,
  // Dropdown,
  FormRow
} from '../../src/components'

const Center = styled.div`
  display: flex;
  ${'' /* justify-content: center; */}
  ${'' /* > div {
    max-width: 60rem;
  } */}
  .dnb-h2,
  .dnb-h3 {
    margin: 0;
    line-height: 1.5rem;
  }
`

export default [
  'FormRow',
  () => (
    <Center>
      <Wrapper showOverflow>
        <Box>
          <H2 bottom="large">Combine vertical and horizontal</H2>
          <FormRow
            label={
              <H3 top={false} bottom="large">
                Custom vertical legend:
              </H3>
            }
            vertical
            top="medium"
            // no_fieldset
          >
            <Input label="Vertical input A" />
            <Input label="Vertical input B" top="medium" />
            <FormRow
              // label="Horizontal legend:"
              // direction="horizontal"
              // indent
              // indent_offset="medium"

              vertical="false"
              label_direction="horizontal"
              top="medium"
            >
              <Input label="Horizontal input A" right="small" />
              <Input label="Horizontal input B" />
            </FormRow>
            <Input label="Vertical input C" top="medium" />
          </FormRow>
        </Box>

        <Box>
          <H3 bottom="large">Plain</H3>
          <AllComponents showText horizontal vertical />
        </Box>

        <Box>
          <H3 bottom="large">Horizontal label</H3>
          <FormRow
            // indent
            // indent_offset="large"
            label="Horizontal Legend Aptent maecenas non pharetra libero massa auctor pretium vulputate vivamus:"
            direction="horizontal"
          >
            <AllComponents showText horizontal />
          </FormRow>
        </Box>

        <Box>
          <H3 bottom="large">Vertical direction</H3>
          <FormRow label="Vertical direction:" direction="vertical">
            <AllComponents showText />
          </FormRow>
        </Box>

        <Box>
          <H3 bottom="large">Vertical everything</H3>
          <FormRow label="Vertical everything:" vertical="true">
            <AllComponents showText />
          </FormRow>
        </Box>

        <Box>
          <H3 bottom="large">Vertical label direction</H3>
          <FormRow label="Vertical Legend:" label_direction="vertical">
            <AllComponents showText horizontal />
          </FormRow>
        </Box>

        <Box>
          <H3 bottom="large">Vertical label direction, but no label</H3>
          <FormRow label="Vertical Legend:" label_direction="vertical">
            <AllComponents showText horizontal hideLabel />
          </FormRow>
        </Box>

        <Box>
          <FormRow label="Inputs legend:">
            <Input value="Input value A ..." right="small" />
            <Input value="Input value B ..." />
          </FormRow>
        </Box>
        <Box>
          <FormRow direction="horizontal" indent="default">
            <FormLabel for_id="alone-1">
              A long horizontal FormLabel with a lot of informative text
              and a default indent:
            </FormLabel>
            <Checkbox id="alone-1" label="Checkbox" />
          </FormRow>
        </Box>
        <Box>
          <FormRow
            direction="horizontal"
            indent
            section_style="mint-green"
            section_spacing="large"
          >
            <FormLabel for_id="alone-2">
              A long horizontal FormLabel with a lot of informative text
              and a default indent:
            </FormLabel>
            <Radio.Group
              id="alone-2"
              // label="Group:"
              // label="Long Group name Vitae dapibus eros viverra torquent euismod at dignissim vel mattis:"
              on_change={({ value }) => {
                console.log('on_change', value)
              }}
              value="first"
              // disabled
              // name="MyGroup" // The Group Name
            >
              <Radio label="First" value="first" />
              <Radio label="Second" value="second" />
              <Radio
                label="Third"
                value="third"
                // checked
              />
            </Radio.Group>
          </FormRow>
        </Box>
      </Wrapper>
    </Center>
  )
]
