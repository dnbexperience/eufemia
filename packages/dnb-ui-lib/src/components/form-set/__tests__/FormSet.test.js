/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../FormSet'
import FormRow from '../../form-row/FormRow'

const props = fakeProps(require.resolve('../FormSet'), {
  optional: true
})
props.direction = 'horizontal'
props.element = 'form'

describe('FormSet component', () => {
  const Comp = mount(<Component {...props} />)

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('should have .dnb-form-set class', () => {
    expect(Comp.find('.dnb-form-set').exists()).toBe(true)
  })

  it('should have working provider with vertical direction class on form-row', () => {
    const Comp = mount(
      <Component {...props} direction="vertical">
        <FormRow />
      </Component>
    )
    expect(
      Comp.find('.dnb-form-row').hasClass('dnb-form-row--vertical')
    ).toBe(true)
  })

  it('should have working provider have correct indent classs on form-row', () => {
    const Comp = mount(
      <Component {...props} indent="large">
        <FormRow />
      </Component>
    )
    expect(
      Comp.find('.dnb-form-row').hasClass('dnb-form-row__indent--large')
    ).toBe(true)
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('FormSet scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-form-set.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-form-set-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
