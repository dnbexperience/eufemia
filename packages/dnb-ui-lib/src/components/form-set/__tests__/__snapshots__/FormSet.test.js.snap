// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSet component have to match snapshot 1`] = `
<FormSet
  class="class"
  className="className"
  custom_element={Object {}}
  custom_method={[Function]}
  direction="horizontal"
  disabled="disabled"
  element="form"
  id="id"
  no_form="no_form"
  prevent_submit="prevent_submit"
  render_content={[Function]}
>
  <Element
    className="dnb-form-set className class"
    is="form"
    onSubmit={[Function]}
  >
    <form
      className="dnb-form-set className class"
      onSubmit={[Function]}
    >
      children
    </form>
  </Element>
</FormSet>
`;

exports[`FormSet scss have to match default theme snapshot 1`] = `
"/*
 * FormSet theme
 *
 */
"
`;

exports[`FormSet scss have to match snapshot 1`] = `
"/*
* DNB FormSet
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-form-set {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: var(--font-size-basis);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-form-set *,
  .dnb-form-set ::before,
  .dnb-form-set ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-form-set ::before,
  .dnb-form-set ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
 * FormSet component
 *
 */
"
`;
